extensionsGallery-insider.json

```diff
@@ -4316,6 +4316,59 @@
 					],
 					"statistics": [],
 					"flags": "preview"
+				},
+				{
+					"extensionId": "85",
+					"extensionName": "net-core-3-runtime",
```

Comment:
It is deliberately tied to 3.1 - this is an LTS release and future releases may or may not be backward compatible, so consumers will explicitly opt-in to a version of runtime they know they work with. We will have separate extension for .NET 5/6 when needed and consumers can decide what they want to use.

Reply:
It would be much preferred not to clutter up our gallery with a bunch of one-off extensions. I'm not sure how backwards compatibility would factor into this - the idea is that the extension should allow an extension to specify what version of the framework it requires and then the extension would validate those exist.

Reply:
@Charles-Gagnon 

> the idea is that the extension should allow an extension to specify what version of the framework it requires and then the extension would validate those exist.

Version-wise, we should always be on latest and greatest, unless there are legitimate reason not to do so. If "version" was referring to the runtime, then this "idea" breaks basic single responsibility principle. Different runtimes have different release cadence. For example, if one extension was to download different runtimes, then whenever we need to upgrade, let's say .NET 5 to latest version - we would "invalidate" .NET Core 3.1 runtime and would have to redownload exactly the same version again by the new version of the extension. There are of course more benefits to componentized model, you can read more about it on the internet.

> not to clutter up our gallery with a bunch of one-off extensions

Can you please expand on this statement? What do you mean by "one-off extensions"? This is an extension with clear scope and responsibility. It will be supported moving forward up until .NET Core 3.1 is deprecated and officially replaced with .NET 5. In fact, I already plan to roll out a new version of this extension with latest 3.1.19 runtime once we get closer to publishing other dependent extensions, because there were multiple security patches that we have to take:

![image](https://user-images.githubusercontent.com/11728682/134072429-dd3c1573-1cbe-487a-9136-f94d6db2078d.png)